Classes/Setup.php

```diff
@@ -163,28 +163,38 @@ public static function execute()
      */
     public static function schema(array $sql) : array
     {
-        $tables = [];
-        $conn = self::context()->db();
+        $ctx = self::getContext();
+        $dbm = $ctx->getDatabaseManager();
+        $connectionNames = array_keys($ctx->getConfig()->get( 'resource'));
+        $connectionNames = array_filter($connectionNames, fn (string $key): bool => str_starts_with($key, 'db'));
 
-        foreach (['fe_users_', 'madmin_', 'mshop_'] as $prefix) {
-            $result = $conn->create('SHOW TABLES like \'' . $prefix . '%\'')->execute();
+        foreach ($connectionNames as $connectionName) {
+            $conn = $dbm->acquire($connectionName);
 
-            while (($row = $result->fetch(\Aimeos\Base\DB\Result\Base::FETCH_NUM)) !== null) {
-                $tables[] = $row[0];
+            $tables = [];
+
+            foreach(['fe_users_', 'madmin_', 'mshop_'] as $prefix) {
+                $result = $conn->create('SHOW TABLES like \'' . $prefix . '%\'')->execute();
+
+                while(($row = $result->fetch(\Aimeos\Base\DB\Result\Base::FETCH_NUM)) !== null) {
+                    $tables[] = $row[0];
+                }
             }
-        }
 
-        foreach ($tables as $table) {
-            $result = $conn->create('SHOW CREATE TABLE ' . $table)->execute();
+            foreach($tables as $table) {
+                $result = $conn->create('SHOW CREATE TABLE `' . $table . '`')->execute();
 
-            while (($row = $result->fetch(\Aimeos\Base\DB\Result\Base::FETCH_NUM)) !== null) {
-                $str = preg_replace('/,[\n ]*CONSTRAINT.+CASCADE/', '', $row[1]);
-                $str = preg_replace('/ DEFAULT CHARSET=[^ ;]+/', '', $str);
-                $str = preg_replace('/ COLLATE=[^ ;]+/', '', $str);
-                $str = str_replace('"', '`', $str);
```

Comment:
You need to keep that lines

Reply:
The adjustment worked fine for us in v11.5 LTS. What's the reason to keep the lines?

Reply:
There had been installations which failed to upgrade when the default charset and collation was part of the output.

Reply:
I'll adjust and test in our setup.

Reply:
TYPO3 v11.5 will drop the foreign key when revmoing the whole key:
```sql
 ALTER TABLE `mshop_supplier_list` DROP FOREIGN KEY `fk_mssupli_pid`
 ```

I therefore kept the removal of charset and collatin, but reduced the removal of the constraint/cascade to the necessary portion.
This extension is marked to be compatible with 11.5 only, so this should be fine I guess.

Reply:
Is that really related? Fear, that `$str = preg_replace('/CONSTRAINT `[a-zA-Z0-9_-]+` /', '', $str);` will only remove the first part of the constraint and `ON UPDATE CASCADE ON DELETE CASCADE` will be left over and then, TYPO3 will throw an error ...

Reply:
I've executed the code on every single change. TYPO3 v11.5 can properly parse foreign keys, it just can't handle the constraint itself.

Reply:
OK, let's try an see ...

Reply:
You can add tests covering the scenario which you then can execute against different dbms and TYPO3 versions.

Reply:
Help for updating the tests is very welcome! :-)